src/lib/tabwidget/tabwidget.cpp

```diff
@@ -787,8 +787,17 @@ QByteArray TabWidget::saveState()
         if (!webTab)
             continue;
 
-        WebTab::SavedTab tab(webTab);
-        tabList.append(tab);
+        // save state does not support saving restore tab
```

Comment:
I mean save state just saves the url of restore tab (i.e. qupzilla:restore) and not the session data loaded in restore tab or at least the path to loaded session file.

This workaround fixes saving sessions when a restore tab is opened. Another solution is closing restore tab but it can change current index and also sometimes closing restore tab will lead to a window without tab.